src/Std/Time/Duration.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.Internal
+import Std.Time.Time
+
+namespace Std
+namespace Time
+open Internal
+
+/--
+`Instant` represents a place in time with second and nanoseconds precision.
+-/
+structure Instant where
+  second : Second.Offset
+  nano : Nanosecond.Ordinal
+  valid : second.val ≥ 0 ∧ nano.val ≥ 0
+  deriving Repr
+
+/--
+Time duration with nanosecond precision. This type allows negative duration.
+-/
+structure Duration where
+  second : Second.Offset
+  nano : Nanosecond.Span
```

Comment:
I wonder whether this representation allows for different encodings of the same duration. Are `Duration.mk 0 500000000` and `Duration.mk 1 -500000000` the same thing?